packages/components/src/custom-select-control-v2/types.ts

```diff
@@ -48,6 +57,69 @@ export type CustomSelectProps = {
 	 */
 	value?: string | string[];
 };
+/**
+ * The legacy object structure for the options array.
+ */
+type Option = {
+	key: string;
+	name: string;
+	style?: {};
+	className?: string;
+	__experimentalHint?: string;
+};
+
+/**
+ * The legacy object returned from the onChange event.
+ */
+type OnChangeObject = {
+	selectedItem: Option;
+	highlightedIndex?: number;
+	isOpen?: boolean;
+};
+
+export type LegacyCustomSelectProps = {
```

Comment:
Looking at [the legacy implementation](https://github.com/WordPress/gutenberg/blob/d2450c88f42e22827718ca233a94cb9c5e1b873f/packages/components/src/custom-select-control/index.js), we're missing a few props:

- `__next40pxDefaultSize`
- `className`
- `onMouseOver`
- `onMouseOut`
- `onFocus`
- `onBlur`

Reply:
In the first iteration (https://github.com/WordPress/gutenberg/pull/57000), 
the legacy props were being translated to the new version which used `WordPressComponentProps`, so we didn't get type errors for these (aside from `__next40pxDefaultSize`). 

Is that something that we should look into adding or is it preferred to explicitly add just these props?

Reply:
I would probably avoid using `WordPressComponentProps` for the legacy, so that we keep it as close as possible to the original, at least for the moment